### ISSUE TITLE
Provide alternative to using __dirname for bundling with Webpack

### ISSUE THREAD

**Issue by chrmarti** [Azure/azure-sdk-for-node#4054]
The callers of `getPackageJsonInfo` are using `__dirname` for locating their package.json which prevents the code from running when bundled using Webpack where the folder structure and the various package.jsons no longer exist. Instead of dynamically looking up the package name and version, each module could inline these during build time.

https://github.com/Azure/azure-sdk-for-node/blob/a0443627390d6bdee9450bba37294ea69f394c9e/runtime/ms-rest/lib/serviceClient.js#L408

/cc @jrieken

**Comment by chrmarti** [comment]
I discovered that using `require(...)` with a string literal works in the normal (unpacked) and the webpacked case (where the file is just packed as well).

So: https://github.com/Azure/azure-sdk-for-node/blob/a0443627390d6bdee9450bba37294ea69f394c9e/lib/services/subscriptionManagement/lib/subscriptionClient.js#L58

Could use `let packageInfo = require('../package.json')` instead of the `getPackageJsonInfo` helper.

**Comment by amarzavery** [comment]
SDKs for all the ARM based packages will be deprecated pretty soon. Apart from the issue you pointed, lot of other changes were also needed for the sdks to be isomorphic. Hence we have created a brand new repo named https://github.com/azure/azure-sdk-for-js. All the sdks in the repo are in TypeScript and the compiled TypeScript code will work out of the box for node.js and browsers. We have made sure to avoid having major breaking changes. There are some changes in how you import/require the new TS packages. Once the client is instantiated, by and large everything remains the same. I would recommend taking a look at [this migration doc](https://github.com/Azure/azure-sdk-for-js/blob/master/documentation/Migration.md) and feel free to open issues in the js repo.

The equivalent TS client can be found [here](https://github.com/Azure/azure-sdk-for-js/tree/master/packages/%40azure/arm-subscriptions).

**NOTE:**  We are pretty close to publishing the new packages to npm. We are in the final leg of our sprint to get this to the closing line.
